azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -314,6 +330,17 @@ public void run() {
               lastExecutionDirCleanTime = currentTime;
             }
 
+            if (flowMaxRunningTimeInMins > 0 && currentTime - LONG_RUNNING_FLOW_KILLING_INTERVAL_MS > lastLongRunningFlowCleanTime) {
```

Comment:
@mukund-thakur we have flow/job level SLA feature. Maybe that's what you are looking for. Please let me know if that doesn't solve your problems.